topology/encryption.go

```diff
@@ -24,12 +26,27 @@ func NewAESEncryption(key []byte) (*AESEncryption, error) {
 	}, nil
 }
 
-func (ae *AESEncryption) Decrypt(data string) []byte {
-	iv := data[:aes.BlockSize]
-	bytes, _ := hex.DecodeString(data[aes.BlockSize:])
-
-	stream := cipher.NewCTR(ae.block, []byte(iv))
-	dst := make([]byte, len(bytes))
-	stream.XORKeyStream(dst, bytes)
+func (ae *AESEncryption) Decrypt(ct []byte) []byte {
+	iv := ct[:aes.BlockSize]
```

Comment:
I'm fine with it all being hex. It was like that at some point, but checked out most tools and they accepted either as plaintext or as plaintext or hex and was more convenient that way.

Reply:
I was actually trying to find some RFC that can describe an implementation of AES CTR mode to be sure In my thoughts. Generally speaking, IV is not a part of the AES standard (AES is a permutation network first of all). IV is a part of the Modes of Operation standard and appears there is a lot of confusion here. 
But I remember for sure that I always was using everything in hex.